packages/modules/playwright/src/playwright-container.test.ts

```diff
@@ -0,0 +1,189 @@
+import path from "path";
+import { existsSync } from "fs";
+import { PlaywrightContainer } from "./playwright-container";
+
+describe("PlaywrightContainer", () => {
+  jest.setTimeout(180_000);
+
+  const PLAYWRIGHT_IMAGE = "mcr.microsoft.com/playwright:latest";
+  const EXTERNAL_PLAYWRIGHT_SAVE_REPORTS_DIRECTORY = path.resolve(__dirname, "..", "example-reports");
+  const EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY = path.resolve(__dirname, "example-project");
+
+  const SUCCESSFUL_TEST_RUNNING_OUTPUT = "Running";
+  const SUCCESSFUL_TEST_PASSED_OUTPUT = "passed";
+  const SUCCESSFUL_TEST_EXIT_CODE = 0;
+
+  it(`should pass example tests with a dot build in reporter for ${process.arch}`, async () => {
+    const startedPlaywrightBuildInReporterContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    ).start();
+
+    const { output, exitCode } = await startedPlaywrightBuildInReporterContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=dot",
+    ]);
+
+    await startedPlaywrightBuildInReporterContainer.stop();
+
+    expect(output).toContain(SUCCESSFUL_TEST_RUNNING_OUTPUT);
+    expect(output).toContain(SUCCESSFUL_TEST_PASSED_OUTPUT);
+    expect(exitCode).toBe(SUCCESSFUL_TEST_EXIT_CODE);
+  });
+
+  it(`should pass example tests with a line build in reporter for ${process.arch}`, async () => {
+    const startedPlaywrightBuildInReporterContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    ).start();
+
+    const { output, exitCode } = await startedPlaywrightBuildInReporterContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=line",
+    ]);
+
+    await startedPlaywrightBuildInReporterContainer.stop();
+
+    expect(output).toContain(SUCCESSFUL_TEST_RUNNING_OUTPUT);
+    expect(output).toContain(SUCCESSFUL_TEST_PASSED_OUTPUT);
+    expect(exitCode).toBe(SUCCESSFUL_TEST_EXIT_CODE);
+  });
+
+  it(`should pass example tests creating an html reporter for ${process.arch}`, async () => {
+    const externalDestinationReporterPath = path.resolve(EXTERNAL_PLAYWRIGHT_SAVE_REPORTS_DIRECTORY, "index.html");
+
+    const startedPlaywrightContainer = await new PlaywrightContainer(
+      PLAYWRIGHT_IMAGE,
+      EXTERNAL_PLAYWRIGHT_PROJECT_DIRECTORY
+    )
+      .withEnvironment({ PLAYWRIGHT_HTML_REPORT: "test-reports" })
+      .start();
+
+    const { output, exitCode } = await startedPlaywrightContainer.exec([
+      "npx",
+      "playwright",
+      "test",
+      "--reporter=html",
+    ]);
+
+    await startedPlaywrightContainer.saveReporter(
+      {
+        type: "html",
+        outputDir: "test-reports",
```

Comment:
should be outputFolder https://playwright.dev/docs/test-reporters#html-reporter

Reply:
I should remove this option, the container is stateless so why we are interested in where place the report inside the container?

we could go with the default options that playwright bring us

Reply:
refactor here https://github.com/testcontainers/testcontainers-node/pull/726/commits/b6930c480ce6836e15c7204eb7277432f84adeab